src/scrapers/visa-cal.ts

```diff
@@ -443,38 +206,160 @@ class VisaCalScraper extends BaseScraperWithBrowser {
+            this.xSiteId = "09031987-273E-2311-906C-8AF85B17C8D9",
+    */
+    return Promise.resolve('09031987-273E-2311-906C-8AF85B17C8D9');
+    // return this.page.evaluate(() => new Ut().xSiteId);
+  }
+
+  getLoginOptions(credentials: Record<string, string>): LoginOptions {
     return {
       loginUrl: `${LOGIN_URL}`,
       fields: createLoginFields(credentials),
       submitButtonSelector: 'button[type="submit"]',
       possibleResults: getPossibleLoginResults(),
       checkReadiness: async () => waitUntilElementFound(this.page, '#ccLoginDesktopBtn'),
       preAction: this.openLoginPopup,
+      postAction: async () => {
+        try {
+          await waitUntilElementFound(this.page, 'button.btn-close');
+          const currentUrl = await getCurrentUrl(this.page);
+          if (currentUrl.endsWith('site-tutorial')) {
+            await clickButton(this.page, 'button.btn-close');
+          }
+        } catch (e) {
+          const currentUrl = await getCurrentUrl(this.page);
+          if (currentUrl.endsWith('dashboard')) return;
+          throw e;
+        }
+      },
       userAgent: 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36',
     };
   }
 
+  isCardTransactionDetails(result: CardTransactionDetails | CardTransactionDetailsError):
+    result is CardTransactionDetails {
+    return (result as CardTransactionDetails).result !== undefined;
+  }
+
   async fetchData(): Promise<ScaperScrapingResult> {
-    const defaultStartMoment = moment().subtract(1, 'years').add(1, 'day');
+    const defaultStartMoment = moment().subtract(1, 'years').subtract(6, 'months').add(1, 'day');
     const startDate = this.options.startDate || defaultStartMoment.toDate();
     const startMoment = moment.max(defaultStartMoment, moment(startDate));
     debug(`fetch transactions starting ${startMoment.format()}`);
 
-    debug('fetch future debits');
-    const futureDebits = await fetchFutureDebits(this.page);
+    const Authorization = await this.getAuthorizationHeader();
+    const cards = await this.getCards();
+    const xSiteId = await this.getXSiteId();
+
+
+    const accounts = await Promise.all(
+      cards.map(async (card) => {
+        debug(`fetch transactions for card ${card.cardUniqueId}`);
 
-    debug('navigate to transactions page');
-    await this.navigateTo(TRANSACTIONS_URL, undefined, 60000);
+        const nextBillingMonth = moment().add(1, 'month');
+        const months = nextBillingMonth.diff(startMoment, 'months');
 
-    debug('fetch accounts transactions');
-    const accounts = await fetchTransactions(this.page, startMoment, this.options);
+        const allMonthsData: (CardTransactionDetails)[] = [];
+        for (let i = 0; i <= months; i += 1) {
+          const month = nextBillingMonth.clone().subtract(i, 'months');
+          const monthData = await fetchPostWithinPage<CardTransactionDetails | CardTransactionDetailsError>(
```

Comment:
@baruchiro in what case did your get back `CardTransactionDetailsError` ? couldn't reproduce it...

Reply:
It was reproduced on development, when the `Authorization` or the `X-Site-ID` are not correct.

Reply:
It was reproduced on development when the `Authorization` or the `X-Site-ID` are not correct.